### PR TITLE
CART-89 mercury: update mercury

### PR DESCRIPTION
**Pull request by yulujia**
update merecury to cc0807e8377e129945834d292be21a6667a8cbb3 to include
the HG_Cancel() fix.

### CODE REVIEW

File: mercury.spec [hunk 1]
```diff
@@ -8,7 +8,7 @@ Group:		Development/Libraries
 License:	ANL
 URL:		http://mercury-hpc.github.io/documentation/
 Source0:	https://github.com/mercury-hpc/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.bz2
-Patch1:		https://github.com/mercury-hpc/mercury/compare/c68870ffc0409c29eece5ba036c6efd3c22cee41^...v1.0.1.patch
+Patch1:		https://github.com/mercury-hpc/mercury/compare/cc0807e8377e129945834d292be21a6667a8cbb3^...v1.0.1.patch
```
[the remark this answers is not in the document]
Comment (yulujia) [reply mercury.spec]: cc0807e8377e129945834d292be21a6667a8cbb3.patch didn't work, had to use v1.0.1...cc0807e8377e129945834d292be21a6667a8cbb3.patch

File: Dockerfile.ubuntu.18.04 [hunk 2]
```diff
@@ -36,8 +36,8 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
             cmake libboost-dev libevent-dev pkg-config
 
 ARG CACHEBUST=1
-RUN curl -sS -f -L -O ${OFI_ART}/libfabric1_1.7.1%7Erc1-1_amd64.deb; \
-    curl -sS -f -L -O ${OFI_ART}/libfabric-dev_1.7.1%7Erc1-1_amd64.deb ; \
+RUN curl -sS -f -L -O ${OFI_ART}/libfabric1_1.8.0%7Erc1-1_amd64.deb; \
```
Comment (brianjmurrell) [inline Dockerfile.ubuntu.18.04]:
```suggestion
RUN curl -sS -f -L -O ${OFI_ART}/libfabric1_1.8.0-1_amd64.deb; \
```

File: Dockerfile.ubuntu.18.04 [hunk 3]
```diff
@@ -36,8 +36,8 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
...
+    curl -sS -f -L -O ${OFI_ART}/libfabric-dev_1.8.0%7Erc1-1_amd64.deb ; \
```
Comment (brianjmurrell) [inline Dockerfile.ubuntu.18.04]:
```suggestion
    curl -sS -f -L -O ${OFI_ART}/libfabric-dev_1.8.0-1_amd64.deb ; \
```

File: Makefile [hunk 4]
```diff
@@ -2,28 +2,16 @@ NAME    := mercury
 SRC_EXT := bz2
 SOURCE   = https://github.com/mercury-hpc/$(NAME)/releases/download/v$(VERSION)/$(NAME)-$(VERSION).tar.$(SRC_EXT)
 #PATCH1 is reversed for RPMS.
-PATCH1  := https://github.com/mercury-hpc/mercury/compare/c68870ffc0409c29eece5ba036c6efd3c22cee41^...v1.0.1.patch
+PATCH1  := https://github.com/mercury-hpc/mercury/compare/v1.0.1...cc0807e8377e129945834d292be21a6667a8cbb3.patch
```
Comment (brianjmurrell) [inline Makefile]: This actually isn't needed any more.

File: Makefile [hunk 5]
```diff
@@ -2,28 +2,16 @@ NAME    := mercury
...
-	0005-Update-CMake-policy-for-CMake-3.12-and-above.patch \
-	0004-HG-Core-fix-potential-race-when-forcing-a-handle-to-.patch \
-	0003-Update-travis-CI-to-use-latest-libfabric-CMake-MPICH.patch \
-	0002-NA-OFI-fix-cancelation-of-operations-that-cannot-be-.patch \
-	0001-HG-fix-cancelation-of-HG-operations-fix-267.patch
+PATCHES := $(shell rm -rf _topdir/BUILD) v1.0.1...cc0807e8377e129945834d292be21a6667a8cbb3.patch
 else
-PATCHES := c68870ffc0409c29eece5ba036c6efd3c22cee41^...v1.0.1.patch
+PATCHES := v1.0.1...cc0807e8377e129945834d292be21a6667a8cbb3.patch
 endif
```
Comment (brianjmurrell) [inline Makefile]: I think all of this conditional processing:
https://github.com/daos-stack/mercury/blob/3042937f2c20cae20cc3d2bbac17c791ae6c23d7/Makefile#L6-L12
can be replaced with a simple:
```
PATCHES := v1.0.1...cc0807e8377e129945834d292be21a6667a8cbb3.patch
```
now that the patch is a more straightforward patch.

**Pull request merged**